libcontainer/cgroups/fs/devices.go

```diff
@@ -26,7 +26,11 @@ func (s *DevicesGroup) Apply(d *cgroupData) error {
 }
 
 func (s *DevicesGroup) Set(path string, cgroup *configs.Cgroup) error {
-	if system.RunningInUserNS() {
+	pids, err := cgroups.GetPids(path)
+	if err != nil {
+		return err
+	}
+	if system.RunningInUserNS(pids[0]) {
```

Comment:
@cyphar 
I'm assuming pid cgroups subsystem which was added but that was resource restriction, apart from that I'm not able to recollect anything from configs.Cgroup for pids.